src/OrbitGl/App.cpp

```diff
@@ -2481,10 +2482,47 @@ Future<std::vector<ErrorMessageOr<CanceledOr<void>>>> OrbitApp::LoadAllSymbols()
 
     loading_futures.push_back(RetrieveModuleAndLoadSymbols(module));
   }
+  if (absl::GetFlag(FLAGS_auto_frame_track)) {
+    // Orbit will try to add the default frame track while loading all symbols.
+    std::ignore = AddDefaultFrameTrack();
+  }
 
   return orbit_base::WhenAll(absl::MakeConstSpan(loading_futures));
 }
 
+Future<ErrorMessageOr<void>> OrbitApp::AddDefaultFrameTrack() {
+  const std::filesystem::path default_auto_preset_folder_path =
+      orbit_base::GetExecutableDir() / "autopresets";
+  const std::filesystem::path stadia_default_preset_path =
+      default_auto_preset_folder_path / "stadia-default-frame-track.opr";
+
+  std::vector<std::filesystem::path> auto_preset_paths = {stadia_default_preset_path};
+
+  // Each preset in auto_preset_paths contains a FrameTrack that users might be interested in
+  // loading by default. Orbit will try to load automatically the first loadable preset from the
+  // list. If no presets could be loaded, Orbit will just log an error.
+  for (std::filesystem::path preset_path : auto_preset_paths) {
+    const ErrorMessageOr<PresetFile> preset = ReadPresetFromFile(preset_path);
+    // Errors on reading a preset from a file won't be shown, Orbit simply will try the next preset
+    // from the list until someone is loadable.
+    if (preset.has_value() &&
+        GetPresetLoadState(preset.value()).state == orbit_data_views::PresetLoadState::kLoadable) {
+      orbit_base::ImmediateExecutor immediate_executor{};
+      return LoadPreset(preset.value())
+          .Then(&immediate_executor, [](ErrorMessageOr<void> result) -> ErrorMessageOr<void> {
+            if (result.has_value()) {
+              ORBIT_LOG("The default frame track was automatically added.");
+            }
+            return result;
+          });
+    }
+  }
+  std::string error_message =
+      "It was not possible to add a frame track automatically, because its module can't be loaded.";
```

Comment:
Isn't this error message possibly misleading? Is that really the only reason why no frame track could be added? What if the preset files cannot be found or cannot be opened / read?

Reply:
Yes, it's true. Thanks for pointing it out. I didn't want to mention the presets here, since this is exposing implementation details somehow, but as potentially users might be able to change it, I can do it.